### PR TITLE
refactor(server): device info service

### PR DESCRIPTION
**Pull request by jrasm91**
Refactor the device info endpoints to be a unified "upsert".

TODO:
- Update the mobile app to use the new endpoint (help?)

### CODE REVIEW

File: server/apps/immich/src/api-v1/device-info/device-info.controller.ts [hunk 1]
```diff
@@ -14,19 +13,23 @@ import { DeviceInfoResponseDto } from './response-dto/create-device-info-respons
...
+  /** @deprecated */
   @Post()
-  async createDeviceInfo(
-    @Body(ValidationPipe) createDeviceInfoDto: CreateDeviceInfoDto,
-    @GetAuthUser() authUser: AuthUserDto,
-  ): Promise<DeviceInfoResponseDto> {
-    return this.deviceInfoService.create(createDeviceInfoDto, authUser);
+  public async createDeviceInfo(@GetAuthUser() user: AuthUserDto, @Body(ValidationPipe) dto: UpsertDeviceInfoDto) {
+    const deviceInfo = await this.deviceInfoService.upsert({ ...dto, userId: user.id });
+    return mapDeviceInfoResponse(deviceInfo);
```
Comment (zoodyy) [inline server/apps/immich/src/api-v1/device-info/device-info.controller.ts:20]: we could simply do `return upsertDeviceInfo(user, dto)` right?

File: server/apps/immich/src/api-v1/device-info/device-info.controller.ts [hunk 2]
```diff
@@ -14,19 +13,23 @@ import { DeviceInfoResponseDto } from './response-dto/create-device-info-respons
...
+  /** @deprecated */
   @Patch()
-  async updateDeviceInfo(
-    @Body(ValidationPipe) updateDeviceInfoDto: UpdateDeviceInfoDto,
-    @GetAuthUser() authUser: AuthUserDto,
-  ): Promise<DeviceInfoResponseDto> {
-    return this.deviceInfoService.update(authUser.id, updateDeviceInfoDto);
+  public async updateDeviceInfo(@GetAuthUser() user: AuthUserDto, @Body(ValidationPipe) dto: UpsertDeviceInfoDto) {
+    const deviceInfo = await this.deviceInfoService.upsert({ ...dto, userId: user.id });
+    return mapDeviceInfoResponse(deviceInfo);
```
Comment (zoodyy) [inline server/apps/immich/src/api-v1/device-info/device-info.controller.ts:27]: we could simply do `return upsertDeviceInfo(user, dto)` right?

File: server/apps/immich/src/api-v1/device-info/device-info.controller.ts [hunk 3]
```diff
@@ -14,19 +13,23 @@ import { DeviceInfoResponseDto } from './response-dto/create-device-info-respons
...
+  }
+
+  @Put()
+  public async upsertDeviceInfo(@GetAuthUser() user: AuthUserDto, @Body(ValidationPipe) dto: UpsertDeviceInfoDto) {
```
Comment (zoodyy) [inline server/apps/immich/src/api-v1/device-info/device-info.controller.ts:31]: this function signature does not have a return type. yet it returns deviceInfo

**Review by fyfrey — COMMENTED** [review]
I like the code-simplification / cleanup of redundant stuff!
I've changed the usage in the dart app to upsert. The app now compiles again.
Left a few remarks, nothing super important.

By the way, I think even old client should still work.

File: server/apps/immich/src/api-v1/device-info/device-info.controller.ts (hunk 1 above)
Comment (jrasm91) [reply server/apps/immich/src/api-v1/device-info/device-info.controller.ts:20]: Sweet, I think you're right.

**Comment by jrasm91** [comment]
@zoodyy The old client would stop working if we removed the post/path endpoints, right? I figured it's easy enough to mark them as deprecated and remove them in a subsequent release.

File: server/apps/immich/src/api-v1/device-info/device-info.controller.ts (hunk 3 above)
Comment (jrasm91) [reply server/apps/immich/src/api-v1/device-info/device-info.controller.ts]: Added the response dto to all three methods.

**Comment by zoodyy** [comment]
@jrasm91 yes, the old client would fail if the old endpoints were removed.
I think we will remove the now deprecated endpoints in some future release to give the users enough time to update the app.

**Pull request merged**